src/lib/models/parsePublicAccess.ts

```diff
@@ -0,0 +1,35 @@
+export type PublicAccessPart = {
+    read: boolean
+    write: boolean
+}
+export type PublicAccess = {
+    metadata: PublicAccessPart
+    data: PublicAccessPart
+}
+
+const parseAccessPart = (accessPart: string): PublicAccessPart => {
+    const canRead = accessPart[0] === 'r'
+    return {
+        read: canRead,
+        write: canRead && accessPart[1] === 'w',
```

Comment:
`canRead` - check is actually superfluous. A string like `-w-w----` wouldn't match regex below. But I think it makes sense to do a check here - both for clarity, and if this function would be used by another other function later. Since it wouldn't be correct without the context of the function below.